packages/table/preview/index.tsx

```diff
@@ -397,30 +426,60 @@ class MutableTable extends React.Component<{}, IMutableTableState> {
     }
 
     private renderNumberSelectMenu(label: string, stateKey: keyof IMutableTableState, values: number[]) {
-        const selectedValue = this.state[stateKey] as number;
+        return this.renderSelectMenu(
+            label,
+            stateKey,
+            values,
+            this.toValueLabel,
+            this.handleNumberStateChange,
+        );
+    }
+
+    private renderSelectMenu<T>(
+        label: string,
+        stateKey: keyof IMutableTableState,
+        values: T[],
+        generateValueLabel: (value: any) => string,
+        handleChange: IMutableStateUpdateCallback,
+    ) {
+        // need to explicitly cast generic type T to string
+        const selectedValue = this.state[stateKey].toString();
         const options = values.map((value) => {
             return (
-                <option key={value} value={value}>
-                    {value}
+                <option key={value.toString()} value={value.toString()}>
+                    {generateValueLabel(value)}
                 </option>
             );
         });
         return (
             <label className="pt-label pt-inline tbl-select-label">
                 {label}
                 <div className="pt-select">
-                    <select onChange={this.updateNumberState(stateKey)} value={selectedValue}>
+                    <select onChange={handleChange(stateKey)} value={selectedValue}>
                         {options}
                     </select>
                 </div>
             </label>
         );
     }
 
+    // Select menu - label generators
+    // ==============================
+
+    private toCellContentLabel(cellContent: CellContent) {
+        if (cellContent === "CellNames") { return "Cell names"; }
+        if (cellContent === "Empty") { return "Empty"; }
+        if (cellContent === "LongText") { return "Long text"; }
```

Comment:
Given that I need to make way for an upcoming usage that will map `TruncatedPopoverMode` values to `string` labels, maybe I'll just change `CellContent` to a number-based `enum` too.